src/workerd/jsg/modules-new.h

```diff
@@ -211,13 +213,21 @@ class Module {
     EVAL = 1 << 2,
   };
 
-  // The EvalCallback is used to to ensure evaluation of a module outside of an
-  // IoContext, when necessary. If the EvalCallback is not set, then the
-  // Flag::EVAL on a module is ignored. If the EvalCallback is set, then any
-  // Modules that have the Flag::EVAL set will have their evaluation deferred
-  // to this callback.
-  using EvalCallback = Function<jsg::Promise<Value>(
-      const Module& module, v8::Local<v8::Module> v8Module, const CompilationObserver& observer)>;
+  // The Evaluator is used to to ensure evaluation of a module outside of an
+  // IoContext, when necessary.
+  class Evaluator final {
```

Comment:
Reviewer note: This is an API improvement. Previously we were passing a `kj::Maybe` reference to a callback which was a bit messy. Using the `Evaluator` class here just improves the ergonomics.